lib/util/template/index.ts

```diff
@@ -201,15 +206,17 @@ export function compile(
   input: CompileInput,
   filterFields = true
 ): string {
-  const data = { ...GlobalConfig.get(), ...input };
+  const env = getChildEnv({}) ?? {};
+  const data = { ...GlobalConfig.get(), ...input, env };
   const filteredInput = filterFields ? proxyCompileInput(data) : data;
   logger.trace({ template, filteredInput }, 'Compiling template');
   if (filterFields) {
     const matches = template.matchAll(templateRegex);
+    const allowedFields = new Set([...allowedFieldsList, ...Object.keys(env)]);
```

Comment:
Are you sure we need to add every key within `env` to the allowed fields list? I was expecting that we just need to add `env` itself

Reply:
When the variable is `env.HOME`, the loop actually checks both the `env` and `HOME` strings, because extracted template variables are split with `match[1].split('.')`:

```
> match[1]
"env.HOME"

> const varNames = match[1].split('.');
["env", "HOME"]
```

So actually an info gets logged for `env` because it is not in `allowedFields`; for `HOME`, it's fine. This is a bug I should fix I think. :bug: 

List `allowedFieldsList` is used when `filterFields == true`, but no actual filtering happens, only messages 'Disallowed variable name in template' are getting logged - is that on purpose? :thinking: 

---

Coming back to your question, I see two solutions:

1. I can add an extra check to see of `varNames[0] == 'env'`;
2. I can add string `env` to list `allowedFields`.

Which one do you think is better?

Reply:
We should allow everything under `env`.

The log message is intentional - the goal was to make it a "soft" warning until we get all the fields in use and only later make it a hard blocking.

Reply:
we already hard block that
https://github.com/renovatebot/renovate/blob/9f0a335fba39f4fb809105dd0ae187ea449ebf6f/lib/util/template/index.ts#L172-L174

Reply:
> We should allow everything under `env`.

I believe [files#diff-29b93fR217](https://github.com/renovatebot/renovate/pull/19301/files#diff-29b93fd71dde6f977077479154849d32614311d877dd83dbf1f6903401ee8088R217) now allows everything under `env`.